spark3-extensions/src/main/antlr/org.apache.spark.sql.catalyst.parser.extensions/IcebergSqlExtensions.g4

```diff
@@ -69,7 +69,18 @@ statement
     : CALL multipartIdentifier '(' (callArgument (',' callArgument)*)? ')'                  #call
     | ALTER TABLE multipartIdentifier ADD PARTITION FIELD transform (AS name=identifier)?   #addPartitionField
     | ALTER TABLE multipartIdentifier DROP PARTITION FIELD transform                        #dropPartitionField
-    | ALTER TABLE multipartIdentifier WRITE ORDERED BY order                                #setTableOrder
+    | ALTER TABLE multipartIdentifier WRITE
+        writeDistributionSpec?
```

Comment:
I decided to allow such calls:

```
ALTER TABLE t WRITE
DISTRIBUTED BY PARTITION
UNORDERED
```

or

```
ALTER TABLE t WRITE
DISTRIBUTED BY PARTITION
UNORDERED
```

Reply:
In both cases we will infer a local sort by partition spec, though.

Reply:
Do we need to require `distributed by` before `ordered by`?

I think it would be more natural to use `WRITE UNORDERED DISTRIBUTED BY PARTITION`. In general, I think it makes sense to avoid order requirements for orthogonal clauses.

Reply:
I was interpreting DISTRIBUTED BY as a stronger clause. For example, WRITE DISTRIBUTED BY ... ORDERED BY ... is going to be translated into a clustered distribution. And since we first satisfy the required distribution, it felt natural to me. However, `WRITE DISTRIBUTED BY UNORDERED` does seem a bit weird.

Reply:
@rdblue, are you proposing to allow any order?

Reply:
Actually, since we allow `WRITE DISTRIBUTED BY PARTITION` without any ordering, maybe it is not that big a deal.

Reply:
Yeah, I think we should add another rule with the opposite order. People shouldn't need to remember which one comes first.

Reply:
I went for the approach used in `visitCreateTableClauses` in Spark and added tests for different combinations.